README.md

```diff
@@ -1,189 +1,325 @@
+By default, each list query will return page 1 of the first 100 results (please see the official [API Documentation](https://www.quandl.com/docs/api) for more detail).
 
-Which outputs:
+You can also retrieve the dataset through the database by using the helper method.
 
+```python
+quandl.Database.get('WIKI').datasets()
 ```
-No authentication tokens found: usage will be limited.
-See www.quandl.com/api for more information.
-Returning Dataframe for  [u'GOOG.NASDAQ_AAPL.4', u'GOOG.NASDAQ_MSFT.4']
 
-        GOOG.NASDAQ_AAPL - Close  GOOG.NASDAQ_MSFT - Close
-Date                                                          
-1997-08-20        6.16                     17.57
-1997-08-21        6.00                     17.23
-1997-08-22        5.91                     17.16
-1997-08-25        5.77                     17.06
-1997-08-26        5.56                     16.88
+### Database Bulk Download
+
+To get the url for bulk download of all datasets data of a database:
 
+```python
+import quandl
+quandl.ApiConfig.api_key = 'tEsTkEy123456789'
+quandl.Database.get('ZEA').bulk_download_url()
+=> "https://www.quandl.com/api/v3/databases/ZEA/data?api_key=tEsTkEy123456789"
 ```
 
+To bulk download all datasets data of a database:
 
-## Push Example
-You can now upload your own data to Quandl through the Python package.
+```python
+quandl.ApiConfig.api_key = 'tEsTkEy123456789'
+quandl.Database.get('ZEA').bulk_download_to_file('/path/to/destination/folder')
+```
 
-At this time the only accepted format is a date indexed Pandas DataSeries.
+For bulk download of premium databases, please ensure that a valid `api_key` is set, as authentication is required.
 
-Things to do before you upload:
+For both `bulk_download_url` and `bulk_download_to_file`, an optional `download_type` parameter can be passed in:
 
-* Make an account and set your authentication token within the package with the Quandl.auth() function.
-* Get your data into a data frame with the dates in the first column.
-* Pick a code for your dataset - only capital letters, numbers and underscores are acceptable.
+```python
+quandl.Database.get('ZEA').bulk_download_to_file('.', download_type='partial')
+```
+
+If `download_type` is not specified, a `complete` bulk download will be performed. Please see the [API Documentation](https://www.quandl.com/docs/api) for more detail.
 
-Then call the following to push the data:
+### MergedDataset
+
+You can get a merged representation of multiple datasets.
 
 ```python
-Quandl.push(data, code='TEST', name='Test', desc='test')
+import quandl
+merged_dataset = quandl.MergedDataset.get([('GOOG/NASDAQ_AAPL', {'column_index': [4]}),
+                                           ('GOOG/NASDAQ_MSFT', {'column_index': [1,4]}), 'GOOG/NYSE_TWTR'])
 ```
 
-All parameters but desc are necessary
+In the above example, column 4 of dataset 'GOOG/NASDAQ_AAPL', columns 1 and 4 of dataset 'GOOG/NASDAQ_MSFT', and all columns of 'GOOG/NYSE_TWTR' is requested.
```

Comment:
This sentence reads a bit strangely. `In the above example, column 4 of dataset 'GOOG/NASDAQ_AAPL', columns 1 and 4 of dataset 'GOOG/NASDAQ_MSFT', and all columns of 'GOOG/NYSE_TWTR' is requested.`

Perhaps try:

In the above example the following data will be merged together:

* column 4 of dataset 'GOOG/NASDAQ_AAPL'
* columns 1 and 4 of dataset 'GOOG/NASDAQ_MSFT'
* and all columns of 'GOOG/NYSE_TWTR'